WooCommerce/Classes/Authentication/AuthenticationManager.swift

```diff
@@ -258,22 +261,22 @@ extension AuthenticationManager: WordPressAuthenticatorDelegate {
     /// Presents the Login Epilogue, in the specified NavigationController.
     ///
     func presentLoginEpilogue(in navigationController: UINavigationController, for credentials: AuthenticatorCredentials, onDismiss: @escaping () -> Void) {
+        let matcher = ULAccountMatcher()
+        matcher.refreshStoredSites()
+
+        /// Jetpack is required. Present an error if we don't detect a valid installation for a self-hosted site.
+        if let urlFromCredentials = credentials.wpcom?.siteURL ?? credentials.wporg?.siteURL,
+           isJetpackValidForSelfHostedSite(url: urlFromCredentials) {
+            return presentJetpackError(for: urlFromCredentials, with: credentials, in: navigationController, onDismiss: onDismiss)
+        }
+
```

Comment:
In `presentJetpackError`, when Jetpack setup completes, we call `presentLoginEpilogue` again to double check that the site has connected to Jetpack. Moving this check up here will cause the Jetpack error to be displayed again.

I'd suggest resetting `currentSelfHostedSite = nil` in `onJetpackSetupCompletion` block right before calling `presentLoginEpilogue` again to avoid displaying the error screen again.